code/object/objectdock.cpp

```diff
@@ -105,8 +105,13 @@ bool dock_check_find_direct_docked_object(object *objp, object *other_objp)
 
 bool dock_check_find_docked_object(object *objp, object *other_objp)
 {
-	Assert(objp != NULL);
-	Assert(other_objp != NULL);
+	Assert(objp != nullptr && objp->signature > 0);
+	Assert(other_objp != nullptr && other_objp->signature > 0);
```

Comment:
FYI, it is better to have separate assertions so that we know exactly what part of the condition failed.